src/main/java/org/digitalstack/chess/ChessBoard.java

```diff
@@ -6,16 +6,34 @@ public class ChessBoard {
     public static int BOARD_HEIGHT = 7;
 
     private final Pawn[][] pieces;
+    private int blackPawns;
+    private int whitePawns;
 
     public ChessBoard() {
-        pieces = new Pawn[BOARD_WIDTH][BOARD_HEIGHT];
+        pieces = new Pawn[BOARD_WIDTH+1][BOARD_HEIGHT+1];
+        blackPawns = 0;
+        whitePawns = 0;
     }
 
     public void add(Pawn pawn, int xCoordinate, int yCoordinate, PieceColor pieceColor) {
-        throw new UnsupportedOperationException("Need to implement ChessBoard.add()");
+        if(pieces[xCoordinate][yCoordinate] == null && isLegalBoardPosition(xCoordinate, yCoordinate) && blackPawns < 7 && whitePawns < 7 ) {
```

Comment:
Negate this condition and fail fast. You will remove one layer of imbrication this way.